types/phoenix_live_view/view_hook.d.ts

```diff
@@ -0,0 +1,39 @@
+import { CallbackRef, OnReply, ViewHookInterface } from "./hooks";
+import LiveSocket from "./live_socket";
+import View from "./view";
+
+export default class ViewHook implements ViewHookInterface {
+  static makeID(): number;
+  static elementID(el: HTMLElement): any; // DT
+  constructor(view: View, el: HTMLElement, callbacks: any); // DT
+  el: HTMLElement; // DT
+
+  __callbacks: any;
+  __listeners: Set<any>;
+  __isDisconnected: boolean;
+  __attachView(view: any): void;
+  __view: (() => any) | (() => never) | undefined;
+  liveSocket: LiveSocket; // DT
+  __mounted(): void;
+  __updated(): void;
+  __beforeUpdate(): void;
+  __destroyed(): void;
+  __reconnected(): void;
+  __disconnected(): void;
+
+  /**
+   * Binds the hook to JS commands.
+   *
+   * @param {ViewHook} hook - The ViewHook instance to bind.
+   *
```

Comment:
Looks like `js()` has no parameters:
```suggestion
```

Reply:
No. the comment (which was generated from the JavaScript source) is wrong. Here's the JavaScript source:

```
  js(){
    let hook = this
    ...
```

I think I will instead remove the confusing (and in my opinion, wrong) `@param` comment.